### ISSUE TITLE
E721: type comparison is partially working

### ISSUE THREAD

**Issue by pmeier** [PyCQA/pycodestyle#946]
Related to the discussion in #882 if type equality or identity comparisons should be allowed. 

Consider the following snippet run with `pycodestyle==2.6.0` on `python==3.6`:

```python
def foo():
    return 0


def bar():
    return 1


assert type(foo()) == type(bar())
assert type(foo()) is type(bar())

foz = foo()
baz = bar()

assert type(foz) == type(baz)
assert type(foz) is type(baz)
```

```
foobar.py:9:20: E721 do not compare types, use 'isinstance()'
foobar.py:10:20: E721 do not compare types, use 'isinstance()'
```

It complains only for the first two assertions. If I'm not mistaken this means that either:
- type comparisons are allowed, but are not registered properly if one does not compare variables or
- type comparisons should still be not allowed, but are by accident.

**Comment by sigmavirus24** [comment]
In the future, please search **closed and** open issues before creating new ones that are duplicates. The issue right below you on the issue list is literally a duplicate of this: https://github.com/PyCQA/pycodestyle/issues/945

**Comment by pmeier** [comment]
@sigmavirus24 

> In the future, please search **closed and** open issues before creating new ones that are duplicates.

I did. This is how I discovered #882. As far as I understand the discussion there, only workarounds exist for this problem. Thus, I was even more surprised to find that one "styles" above checks fine while the other fails.

> The issue right below you on the issue list is literally a duplicate of this: #945

Maybe it is related, but from my perspective its definitively not a duplicate. #945 is specifically about how to handle `None` for E721, which is not part of the problem I described.
